articles/2016-09-01-semantic-html-tags.md

```diff
@@ -0,0 +1,98 @@
+
+#### Readability
+
+It also makes your code significantly easier to read by other humans. In a complex or large page with levels and levels of `<div>`s, it becomes impossible to understand the flow of information. Too many `<div>`s can fill up a page until it becomes unreadable. This is commonly referred to as _divitis_. Avoid it at all costs.
+
+However, it is much easier to understand if you take advantage of elements like `<main>`, `<section>`, `<header>`, etc. because it helps you visualize the flow of information as you read through the HTML.
+
+#### Styling
+
+An additional benefit of using semantic tags, is that you can write more general CSS selectors. For a refresher on specificity of CSS selectors, read the [blog post](2016-08-02-why-you-should-never-use-important.md) I wrote on the subject). The benefit of general selectors is that you can establish a foundation of styles (for example, all `<section>`s have a blue background and 10 pixels of padding), that can easily be overwritten in a more specific situation, without having to add too many extra classes or incorporating `!important`.
+
+It should be stated that this approach only works in certain cases, usually smaller or simpler UI's. Styling on tag selectors is considered a bad practice if the UI is too large to be appropriately represented as such. There are situations where it can work, but you are more likely going to need to use class selectors for your styling.  
+
+### Frameworks and Templating
+
+If you're using a front-end framework like Ember, you will notice that a lot of components will be wrapped in `<div>`s. This is ok, as the benefit of a `<div>` is that it's generic enough to be used everywhere. If you are committed to using semantic tags, you have two choices here. You can either wrap the template inside the component with the appropriate tag,
+
+```handlebars
+{{#some-component}}
+  <section>
+    <!-- section content goes here -->
+  </section>
+{{/some-component}}
+```
+
+This will render to the following HTML
+
+```html
+<div id="ember123">
+  <section>
+    <!-- section content goes here -->
+  </section>
+</div>
+```
+
+Another option (one that I personally prefer), is to change the `tagName` attribute of the component to match the appropriate semantic tag,
+
+```handlebars
+{{#some-component
+  tagName='section'
+}}
+  <!-- section content goes here -->
+{{/some-component}}
+```
+
+Which will render to
+
+```html
+<section id="ember123">
+  <!-- section content goes here -->
+</section>
+```
+
+The reason I prefer this option is because the wrapper `<div>` in choice 1 isn't actually necessary, and it can get in the way of some of the interesting things you can do with relative pseudo-selectors (for a refresher on what that is, here's the [blog post](2016-05-23-selectors-part-2-pseudo-classes.md) I wrote about it).
+
+## Example
+
+Here is an example of what the outline of a page would look like using semantic HTML5 elements.
+
+```html
```

Comment:
Some simple, fake content would probably make this example more understandable. Might also be nice to have a screen shot of a simply styled version of this markup. The idea being that it gives a clearly picture of the `<nav>` tag being for the top level site navigation, for instance.